### ISSUE TITLE
Add first-class support for `polars`

### ISSUE THREAD

**Issue by maximlt** [holoviz/holoviews#5939]
I know @Hoxbro has that in mind but couldn't find an issue about it. I'm updating hvPlot's docs for the 0.9.0 release and want to mention the support of [Polars](https://www.pola.rs/) isn't as deep as it is with other data types like Dask. Opening this issue to link it from hvPlot's docs.

**Comment by MarcoGorelli** [comment]
Would the task here be to implement `PolarsInterface` in `holoviews/core/data/polars.py`, just like there's `PandasInterface` in `holoviews/core/data/pandas.py`?

**Comment by maximlt** [comment]
Yes I think that's the minimum required. Looking through the code base for `dask` or `cudf` for example reveals some other places where different data types are handled.

**Comment by douglas-raillard-arm** [comment]
Is there any plan to add support for the dataframe protocol ?
https://data-apis.org/dataframe-protocol/latest/index.html
polars seems to implement it, although my understanding is that it would be less efficient than handling polars directly since that API only allows converting between formats rather than accessing the content directly.

**Comment by Hoxbro** [comment]
Maybe in the future. Currently, my plan is:
1) Update pandas DataFrame with MultiIndex support (https://github.com/holoviz/holoviews/pull/6061).
2) Implement a `PolarsInterface`. 
3) Maybe take a look at the dataframe protocol.

**Comment by MarcoGorelli** [comment]
> Implement a PolarsInterface

If you ever wanted to pair on this, feel free to book some time https://calendly.com/marcogorelli

**Comment by hoxbro** [comment]
This should now be supported in HoloViews 1.22, where we added support for Narwhals.
